common/http.go

```diff
@@ -1,60 +1,143 @@
 package common
 
 import (
-	"encoding/json"
+	"context"
 	"fmt"
 	"io"
+	"log/slog"
+	"mime"
 	"net/http"
-	"net/url"
+
+	"github.com/spyzhov/ajson"
 )
 
-func DoHttpGetCall(c *http.Client, baseURL string, endpoint string, token string) (result map[string]interface{}, e *ErrorWithStatus) {
-	url, err := url.JoinPath(baseURL, endpoint)
+// Header is a key/value pair that can be added to a request.
+type Header struct {
+	Key   string
+	Value string
+}
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+	CloseIdleConnections()
+}
+
+type ErrorHook func(rsp *http.Response, body []byte) error
```

Comment:
I wonder if something like ErrorInterpreter or ErrorParser would be more intuitive. Can you also add a comment to this type to explain it more?

Reply:
ErrorHandler maybe? And yeah I'll add some docs.

Reply:
ErrorHandler sounds more like a method that resolves the error somehow, this is more of a translator. ErrorTranslator?